### DISCUSSION TITLE
Re-encode unreliably seekable videos from GUI

### DISCUSSION THREAD

**Discussion by talmo** [talmolab/sleap#1489] [Ideas]
A useful feature that would common in handy quite frequently would be a wrapper to call ffmpeg to reencode unreliably seekable videos.

We currently have a help page that mentions how to do this from the CLI: https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format

**MVP:** GUI option to re-encode project videos (by subprocess calling ffmpeg) and replacing the resulting output automatically within the project.

**Workaround:** Do this in a bash script: https://github.com/talmolab/sleap/issues/710#issuecomment-1104249747

**Related issues**: #710

**Comment by arie-matsliah** [comment]
Can we use one of these approaches to eliminate the need for re-encoding?
https://scikit-image.org/docs/dev/user_guide/video.html

**Comment by talmo** [comment]
Oh wow, great find @ariematsliah-princeton!

This PyAV-based solution would actually be perfect: https://scikit-image.org/docs/dev/user_guide/video.html#adding-random-access-to-pyav

Currently we use opencv-python because it has ffmpeg support linked in, so we can decode in memory without piping to the ffmpeg CLI (super slow). This is implemented here:
https://github.com/murthylab/sleap/blob/1b27526220136d7ec7f41496ca6f5fff1b97c6d9/sleap/io/video.py#L337-L489

We could try switching to PyAV instead, which seems to provide a similar interface AND has binary ffmpeg support. Also seems like it would be pretty much a drop-in replacement. Sounds like the best of all worlds.

We just need to check that it's compatible across platforms and that we can package it with our other deps.

**Comment by arie-matsliah** [comment]
👍 I can give it a try sometime this week and update.
Do we have an easy way to verify if it actually works after the change?
